### PR TITLE
[3.x] Backport jQuery 3.5 security fixes

### PR DESCRIPTION
**Pull request by SniperSister**
The recently released JQuery 3.5.x has fixed two XSS issues. As we can't upgrade to a newer jQuery version in 3.x for BC reasons, the security team suggest to manually patch our jQuery.

As this change might have unexpected b/c implications and the vulnerability already is public and well documented, I'm opening this ticket on the public tracker.

### Summary of Changes
Backported jQuery 3.5 security fixes.

### Testing Instructions
Apply patch, browse various parts of the site and make sure that no JS errors show up in the browser console.

### CODE REVIEW

**Comment by SniperSister** [comment]
There's a potential breaking change in the handling of self-closing tags, see:
https://jquery.com/upgrade-guide/3.5/

**Comment by wilsonge** [comment]
I assume backporting the associated jQuery Migrate fix is beyond us? Because if so we need to really scream about this in the release notes and ideally on social media to extension devs too

**Comment by SniperSister** [comment]
If I understand the JQuery notes correctly, the backport of the compatibility layer re-introduces the issue.

**Comment by zero-24** [comment]
I have just navigated the backend with latest Chrome, Edge and IE11 and also the emulated IE9. Is there anything special we should test here? Do we even use that function patched in core?

**Comment by SniperSister** [comment]
> Do we even use that function patched in core?

That function is used everywhere because it's part of the DOM manipulation library

**Comment by SniperSister** [comment]
@wilsonge I have updated the PR with the "pre-fix-behavior" backport that Drupal published

File: media/jui/js/jquery.js [hunk 1]
```diff
@@ -12,6 +12,7 @@
  * Date: 2016-05-20T17:17Z
  *
  * Modified by Joomla: Mitigate possible XSS vulnerability (gh-2432), CMS Issue 19464; Prevent Object Prototype Polution, https://github.com/jquery/jquery/pull/4333
+ * Modified by Joomla: Mitigate possible XSS vulnerability CVE-2020-11022 and CVE-2020-11023, CMS Issue; https://github.com/DanielRuf/snyk-js-jquery-565129
```
Comment (zero-24) [inline media/jui/js/jquery.js]: Should't we update this to the drupal soure?

**Review by zero-24 — CHANGES_REQUESTED** [review]

File: media/jui/js/jquery.js (hunk 1 above)
Comment (zero-24) [reply media/jui/js/jquery.js]: Or atleast add it somewhere.

Comment (SniperSister) [reply media/jui/js/jquery.js]: done

**Comment by zero-24** [comment]
I have tested this item :white_check_mark: successfully on 4be8cd72c3e8f34924b203876e433e9859fd3aee<br /><br />As mention in the Testing Instructions i have applied the patch, navigated the backend and have not noticed any error message form the browser console. Thanks @SniperSister <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/28948">issues.joomla.org/tracker/joomla-cms/28948</a>.</sub>

**Comment by conseilgouz** [comment]
I have tested this item :white_check_mark: successfully on 4be8cd72c3e8f34924b203876e433e9859fd3aee<br /><br />Just tested it using Chrome. 

I did not see any difference front-end/back-end in browser console.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/28948">issues.joomla.org/tracker/joomla-cms/28948</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/28948">issues.joomla.org/tracker/joomla-cms/28948</a>.</sub>

**Comment by HLeithner** [comment]
Awesome job thanks

**Comment by Webmaster-2020** [comment]
Is it normal to still have old version of jQuery after 3.9.19 update ?

Thank's

**Comment by conseilgouz** [comment]
Bonjour,

En anglais SVP. Messages should be written in english.
Is it normal to still have old version of jQuery after 3.9.19 update ?

The answer is Yes. If you take a look at the header of media/jui/js/jquery.js file, you'll find a comment talking about CVE-2020-11022 which is "the patch". 

Jquery version is now v1.12.4-joomla.

**Comment by Webmaster-2020** [comment]
@pmleconte 
thank you very much for your answer.

**Pull request merged**